Src/Common/WebHeaderCollectionExtensions.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace Microsoft.ApplicationInsights.Common
+{
+    using System;
+    using System.Collections.Specialized;
+    using System.Globalization;
+    using System.Linq;
+
+    /// <summary>
+    /// WebHeaderCollection extension methods.
+    /// </summary>
+    internal static class WebHeaderCollectionExtensions
+    {
+        /// <summary>
+        /// For the given header collection, for a given header of name-value type, find the value of a particular key.
+        /// </summary>
+        /// <param name="headers">Header collection.</param>
+        /// <param name="headerName">Name of the header in the collection.</param>
+        /// <param name="keyName">Desired key of the key-value list.</param>
+        /// <returns>Value against the given parameters.</returns>
+        public static string GetNameValueHeaderValue(this NameValueCollection headers, string headerName, string keyName)
```

Comment:
Are those methods safe and keep the header collection in the same state it was before? I remember issue years ago when one could've touched headers and some flag was set off, so the next attempt to access those headers (from user code) failed. New method of reading headers was introduced in .NET since then, not sure if it applies here, though..